apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte

```diff
@@ -240,6 +246,50 @@
     $apps.open = true;
   }
 
+  const goToNextLesson = () => {
+    const currentLessonIdx = $lessons.findIndex((lesson) => lesson.id === data.lessonId);
+
+    if (currentLessonIdx !== -1 && currentLessonIdx + 1 < $lessons.length) {
+      const nextLesson = $lessons[currentLessonIdx + 1];
+      console.log('next lesson', nextLesson);
+
+      fetchReqData(nextLesson.id, true);
+      const nextLessonUrl =
+        $globalStore.isStudent && !nextLesson.is_unlocked
+          ? $page.url.pathname
+          : `/courses/${$course.id}/lessons/${nextLesson.id}`;
+
+      goto(nextLessonUrl);
+    }
+  };
+
+  const goToPrevLesson = () => {
+    const currentLessonIdx = $lessons.findIndex((lesson) => lesson.id === data.lessonId);
+
+    if (currentLessonIdx !== -1 && currentLessonIdx - 1 >= 0) {
+      const prevLesson = $lessons[currentLessonIdx - 1];
+      console.log('next lesson', prevLesson);
+
+      fetchReqData(prevLesson.id, true);
+      const prevLessonUrl =
+        $globalStore.isStudent && !prevLesson.is_unlocked
+          ? $page.url.pathname
+          : `/courses/${$course.id}/lessons/${prevLesson.id}`;
+
+      goto(prevLessonUrl);
+    }
+  };
+
+  $: {
+    const currentLessonIdx = $lessons.findIndex((lesson) => lesson.id === data.lessonId);
+    disabled.next =
+      currentLessonIdx === -1 ||
+      $lessons.slice(currentLessonIdx + 1).every((lesson) => !lesson.is_unlocked);
+    disabled.prev =
+      currentLessonIdx === -1 ||
+      $lessons.slice(0, currentLessonIdx).every((lesson) => !lesson.is_unlocked);
+  }
```

Comment:
This is a complex logic.

Just delete this reactivity and write one function `isNextOrPrevDisabled`

```svelte
const isNextOrPrevDisabled = (lessonId: string, isPrev: boolean) => {
  const index = $lessons.findIndex((lesson) => lesson.id === lessonId);

  return isPrev ? !$lessons[index - 1] : !$lessons[index + 1]
}

<button
  disabled={isNextOrPrevDisabled(data.lessonId, true)}
>
Prev
</button>

<button
  disabled={isNextOrPrevDisabled(data.lessonId)}
>
Next
</button>
```

Reply:
i needed to handle an extra case when the next lesson being navigated to is currently locked, the function only navigates to the nex function that is not locked